src/index.js

```diff
@@ -447,16 +447,18 @@ export default class HyperScreen extends React.Component {
     // Fetch the resource, then perform the action on the target and undo indicators.
     const fetchAndUpdate = () => this.fetchElement(href, verb, newRoot, formData)
       .then((newElement) => {
+        // If the fetch failed, don't update the doc state.
+        if (!newElement) {
```

Comment:
if we return at this point, doesn't this cause an issue with loading indicators not being cleared properly?

Reply:
Since the fetch was unsuccessful maybe it is ok to not clear the indicators?

Reply:
I don't think that's right, indicators are meant to represent some activity is pending, but we're not auto-retrying calls as I understand it

Reply:
Agreed, this is a known limitation (showing loading state when request fails) for now. Do you have any recommendation for this?

Reply:
I'm not sure what you mean - I think we do hide the loading state (and custom loading indicators) when the request fail right now, don't we?

Reply:
Not in all cases, for example when loading a sectionlist if the next page fails to load we keep showing the spinner. 

https://github.com/Instawork/hyperview/assets/28518512/534b18d7-7bec-4214-9807-b50681b91293

Reply:
Right, I think that's a bug in the component that wraps the section list, which assumes the request will be succesful (the spinner is always visible instead of being shown when the request kicks off). This is not related to the way Hyperview handles loading indicators.

Reply:
Thanks for clarifying. I will put up a follow up pr to clear the loading indicators once I figure out and fix the root cause. But for now I think this is ok because with the previous implementation the app was crashing.